### PR TITLE
add privateCA support for containerd images

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

Adds the ability to support privateCA for containerd images

## Related Issues

https://github.com/astronomer/issues/issues/6022

## Testing

tested on AWS, GCP, Azure, local clusters

## Merging

release-0.34

### CODE REVIEW

**Review by rishkarajgi — COMMENTED** [review]
Please add tests to this change

File: templates/trust-private-ca-on-all-nodes/containerd-ca-update-script.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  labels:
+    component: houston
+    tier: astronomer
+    release: {{ .Release.Name }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: {{ .Release.Service }}
+data:
+  update-containerd-certs.sh: |
+    #!/usr/bin/env sh
+    if [ ! -f /hostcontainerd/config.toml ]; then
```
Comment (rob-1126) [inline templates/trust-private-ca-on-all-nodes/containerd-ca-update-script.yaml:18]: theoretically I think this file could be totally absent and need to be created, but if we have tested on Azure, AWS, and GCP I'm fine with crossing that bridge when we come to it. In reality, I would expect it always exists.

File: values.yaml [hunk 2]
```diff
@@ -19,6 +19,10 @@ global:
   privateCaCertsAddToHost:
     enabled: false
     hostDirectory: /etc/docker/certs.d
+    addToContainerd: false
+    containerdConfigPath: /etc/containerd/certs.d
```
Comment (rob-1126) [inline values.yaml:23]: key name should be something else as that sounds like the path that should put to containerd.toml itself. Maybe rename to containerdCertsConfigPath or similar?

File: values.yaml [hunk 3]
```diff
@@ -19,6 +19,10 @@ global:
...
+    containerdnodeAffinitys: []
```
Comment (rob-1126) [inline values.yaml:24]: Are these actually templated into the daemonset anywhere?

**Review by rob-1126 — CHANGES_REQUESTED** [review]

File: templates/trust-private-ca-on-all-nodes/containerd-ca-update-script.yaml (hunk 1 above)
Comment (pgvishnuram) [reply templates/trust-private-ca-on-all-nodes/containerd-ca-update-script.yaml:18]: yes this file always exists in all cloud provider only difference is the content

**Review by rob-1126 — APPROVED** [review]
Needs to be tested in QA on all three major cloud-providers as part of the release process for both new install and upgrade path.

**Comment by rishkarajgi** [comment]
> Needs to be tested in QA on all three major cloud-providers as part of the release process for both new install and upgrade path.

cc: @himabindu07 @Priyanka13Astro

**Pull request merged**